Consul/DiscoveryChain.cs

```diff
@@ -124,6 +124,36 @@ public class DiscoveryTarget
         public string Name { get; set; }
     }
 
+    public class DiscoveryChainOptions
+    {
+        public string EvaluateInDatacenter { get; set; }
```

Comment:
I think this should not be a part of this structure as it is being used as `[compile-dc](https://developer.hashicorp.com/consul/api-docs/discovery-chain#compile-dc)` which is a query-only parameter.